ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlClient.java

```diff
@@ -43,75 +42,42 @@
-      = "ksql.client.disable.hostname.verification";
-  public static final String TLS_ENABLED_PROP_NAME = "ksql.client.enable.tls";
-
   private final Vertx vertx;
-  private final HttpClient httpClient;
+  private final HttpClient httpNonTlsClient;
+  private final HttpClient httpTlsClient;
   private final LocalProperties localProperties;
   private final Optional<String> basicAuthHeader;
-  private final boolean isTls;
 
   public KsqlClient(
       final Map<String, String> clientProps,
       final Optional<BasicCredentials> credentials,
       final LocalProperties localProperties,
       final HttpClientOptions httpClientOptions
   ) {
-    this(Vertx.vertx(), clientProps, credentials, localProperties, httpClientOptions);
-  }
-
-  @VisibleForTesting
-  KsqlClient(
-      final HttpClient httpClient,
-      final boolean isTls,
-      final Optional<BasicCredentials> credentials,
-      final LocalProperties localProperties
-  ) {
-    this(null, httpClient, isTls, credentials, localProperties);
-  }
-
-  private KsqlClient(
-      final Vertx vertx,
-      final HttpClient httpClient,
-      final boolean isTls,
-      final Optional<BasicCredentials> credentials,
-      final LocalProperties localProperties
-  ) {
-    this.vertx = vertx;
-    this.httpClient = Objects.requireNonNull(httpClient, "httpClient");
-    this.isTls = isTls;
+    this.vertx = Vertx.vertx();
     this.basicAuthHeader = createBasicAuthHeader(
         Objects.requireNonNull(credentials, "credentials"));
     this.localProperties = Objects.requireNonNull(localProperties, "localProperties");
-  }
-
-  private KsqlClient(
-      final Vertx vertx,
-      final Map<String, String> clientProps,
-      final Optional<BasicCredentials> credentials,
-      final LocalProperties localProperties,
-      final HttpClientOptions httpClientOptions
-  ) {
-    this(vertx, createHttpClient(vertx, clientProps, httpClientOptions), httpClientOptions.isSsl(),
-        credentials, localProperties);
+    this.httpNonTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, false);
+    this.httpTlsClient = createHttpClient(vertx, clientProps, httpClientOptions, true);
   }
 
   public KsqlTarget target(final URI server) {
     final boolean isUriTls = server.getScheme().equalsIgnoreCase("https");
-    if (isTls != isUriTls) {
-      throw new KsqlRestClientException("Cannot make request with scheme " + server.getScheme()
-          + " as client is configured " + (isTls ? "with" : "without") + " tls");
-    }
-    return new KsqlTarget(httpClient,
+    final HttpClient client = isUriTls ? httpTlsClient : httpNonTlsClient;
+    return new KsqlTarget(client,
         SocketAddress.inetSocketAddress(server.getPort(), server.getHost()), localProperties,
         basicAuthHeader);
   }
 
   public void close() {
     try {
-      httpClient.close();
+      httpTlsClient.close();
+    } catch (Exception ignore) {
+      // Ignore
```

Comment:
Not directly related to this PR, but is it common practice to ignore exceptions on close? I would've expected to at least log the exception?

Reply:
It's a pretty common approach if you want a clean close.